### PR TITLE
Tests for Elytron mappers and decoders configuration

### PR DESCRIPTION
**Pull request by zebrik**

### CODE REVIEW

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/ElytronPrincipalDecoderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+                    .build())).assertSuccess();
+            aggregatePrincipalDecoderVerifier.verifyExists();
+
+            page.navigateToDecoder().selectResource(AGGREGATE_PRINCIPAL_DECODER_LABEL).getResourceManager()
+                    .removeResource(aggregatePrincipalDecoderName).confirmAndDismissReloadRequiredMessage().assertClosed();
+            assertFalse("Removed resource should not be present in the table any more!",
+                    page.resourceIsPresentInMainTable(aggregatePrincipalDecoderName));
+            aggregatePrincipalDecoderVerifier.verifyDoesNotExist();
+        } finally {
+            ops.removeIfExists(aggregatePrincipalDecoderAddress);
```
Comment (honza-kasik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/ElytronPrincipalDecoderTestCase.java:420]: Shouldn't `constantPrincipalDecoder1address` and `constantPrincipalDecoder2address` be removed here to?

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/ElytronPrincipalDecoderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+                    .build())).assertSuccess();
+            concatenatingPrincipalDecoderVerifier.verifyExists();
+
+            page.navigateToDecoder().selectResource(CONCATENATING_PRINCIPAL_DECODER_LABEL).getResourceManager()
+                    .removeResource(concatenatingPrincipalDecoderName).confirmAndDismissReloadRequiredMessage().assertClosed();
+            assertFalse("Removed resource should not be present in the table any more!",
+                    page.resourceIsPresentInMainTable(concatenatingPrincipalDecoderName));
+            concatenatingPrincipalDecoderVerifier.verifyDoesNotExist();
+        } finally {
+            ops.removeIfExists(concatenatingPrincipalDecoderAddress);
```
Comment (honza-kasik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/ElytronPrincipalDecoderTestCase.java:569]: Shouldn't `constantPrincipalDecoder1address` and `constantPrincipalDecoder2address` be removed here to?

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/LowercaseCustomPrincipalDecoder.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.jboss.hal.testsuite.test.configuration.elytron.principal.decoder;
+
+import java.security.Principal;
+
+import org.wildfly.security.auth.server.PrincipalDecoder;
+
+public class LowercaseCustomPrincipalDecoder implements PrincipalDecoder {
```
Comment (honza-kasik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/LowercaseCustomPrincipalDecoder.java]: IMHO some Javadoc for this class would be nice.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/UppercaseCustomPrincipalDecoder.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.jboss.hal.testsuite.test.configuration.elytron.principal.decoder;
+
+import java.security.Principal;
+
+import org.wildfly.security.auth.server.PrincipalDecoder;
```
Comment (honza-kasik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/principal/decoder/UppercaseCustomPrincipalDecoder.java:5]: Javadoc would be nice here. (Optional)

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/role/decoder/BugsBunnyCustomRoleDecoder.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package org.jboss.hal.testsuite.test.configuration.elytron.role.decoder;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+import org.wildfly.security.authz.AuthorizationIdentity;
+import org.wildfly.security.authz.RoleDecoder;
+import org.wildfly.security.authz.Roles;
+
+public class BugsBunnyCustomRoleDecoder implements RoleDecoder {
```
Comment (honza-kasik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/role/decoder/BugsBunnyCustomRoleDecoder.java]: Some Javadoc here would be nice IMHO.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/role/decoder/SuperMarioCustomRoleDecoder.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package org.jboss.hal.testsuite.test.configuration.elytron.role.decoder;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import org.wildfly.security.authz.AuthorizationIdentity;
+import org.wildfly.security.authz.RoleDecoder;
+import org.wildfly.security.authz.Roles;
+
+public class SuperMarioCustomRoleDecoder implements RoleDecoder {
```
Comment (honza-kasik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/role/decoder/SuperMarioCustomRoleDecoder.java]: Nice class name :+1:. But some Javadoc would be nice :).

**Review by honza-kasik — CHANGES_REQUESTED** [review]
Thanks! I found some minor issues, see comments - missing Javadoc is optional to fix.

**Comment by zebrik** [comment]
@honza-kasik thanks for your input. Fixes are in.

**Review by honza-kasik — APPROVED** [review]

**Comment by honza-kasik** [comment]
Thank you! Merging!

**Pull request merged**